### ISSUE TITLE
AutoModuleShareDirs break `dzil listdeps`. Can't install repo dependencies

### ISSUE THREAD

**Issue by moollaza** [duckduckgo/p5-dzp-automodulesharedirs#4]
Recently I ran into issues running `dzil listdeps` on our zeroclickinfo repos.

After much debugging of our dist.ini files, I realized the commenting out `[AutoModuleShareDirs]` fixes the problem.

It seems that `dzil build` executes the `AutoModuleSharDirs` plugin, which in turn tries to load each Instant Answer module. The issue here is that at this point the user has not installed all the Perl dependencies for the IAs to load (which is the purpose of `dzil listdeps`) and so we have a fatal error:

```
Can't locate Text/Trim.pm in @INC (@INC contains: lib /usr/local/perlbrew/perls/5.16/lib/site_perl/5.16.3/x86_64-linux /usr/local/perlbrew/perls/5.16/lib/site_perl/5.16.3 /usr/local/perlbrew/perls/5.16/lib/5.16.3/x86_64-linux /usr/local/perlbrew/perls/5.16/lib/5.16.3 .) at lib/DDG/Goodie/Base64.pm line 6.
BEGIN failed--compilation aborted at lib/DDG/Goodie/Base64.pm line 6.
Compilation failed in require at /usr/local/perlbrew/perls/5.16/lib/site_perl/5.16.3/x86_64-linux/Moose/Object.pm line 22.
```

I think we need to prevent `dzil install` from executing this plugin, or we need to update this plugin to more gracefully handle classes that fail to load.

---

You can recreate this by uninstall `Text::Trim` (or any other Perl dependency of any Instant Answer) and running `dzil listeps` on the Spice or Goodie repo.

@zachthompson any ideas?

**Comment by moollaza** [comment]
This is now affecting the Spice Repo Travis tests. The tests are failing because I cleared the caches (to force the new DDG module to be installed) and now the dependencies aren't able to be installed.

See https://travis-ci.org/duckduckgo/zeroclickinfo-spice#L300

**Comment by moollaza** [comment]
Fixed in #5
